### PR TITLE
Improve the logic to be much more user friendly

### PR DESCRIPTION
**Pull request by danielufeli**
### What does this PR do?
Add Game, Board, Player Class, and Functionality

### Description of Task to be completed?
- Add Game Class
- Add Player Class
- Add Board Class
- Main Functionality

### CODE REVIEW

**Review by Georjane — CHANGES_REQUESTED** [review]
## Status: Changes Required 🛑 🔧 
### Hi there Team ✋🏽 
Very nice work done. Your game is easy and very simple to understand and play. Great job!
However, we have a few changes to make to improve on our project.

- [ ] No puts/print and gets.chomp outside main.rb file.
- [ ] You no longer need the .gitkeep file in the lib folder
- [ ] All game-logic-related resources (classes and/or modules) are kept in the `lib` directory. But you have a module in the main.rb file
- [ ] The code of the project should be logically divided between classes, but you have all logic in the main file

Please kindly fix these issues and submit for another code review

_Feel free to reach out to me on Slack if you have any questions, Witah Georjane_ 🤝 
Good luck on your project and Happy Coding 💻 

**Witah Georjane
Code Reviewer at Microverse Inc
[portfolio](https://witahgeorjane.tech/) | [linkedIn](https://www.linkedin.com/in/witah-georjane/)**

File: lib/game.rb [hunk 1]
```diff
@@ -10,4 +14,34 @@ def initialize
     @player2 = nil
     @winner = nil
   end
+
+  def valid_move(index)
+    index.between?(1, 9) && @board.board_cell[index - 1].is_a?(Numeric)
+  end
+
+  def move(player)
+    puts "#{player.name}'s turn"
```
Comment (cyonii) [inline lib/game.rb]: - [ ] No `puts/print` and `gets.chomp` outside `main.rb` file. You should find and replace similar usage in your codebase.

**Review by cyonii — CHANGES_REQUESTED** [review]
# Status: Changes Required

Hello team,

You've shown great effort building your project this far, nice job implementing some of the requested changes by the previous reviewer.
- You did a great job with your implementation of OOP 👏

## To Improve
However, there are still minor defects in your code, please take a look at these issues pointed below, kindly fix them and resubmit for another review.
- [ ] No `puts/print` and `gets.chomp` outside `main.rb` file. Please remove those in your `game.rb` and `board.rb` files ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

### You can find me on slack 👉 [CY Kalu](https://microverse-students.slack.com/team/U01494U96FR) if you need further assistance or clarification. I'm always available and will be glad to help.

**Review by enaburekhan — APPROVED** [review]
# Status: Project Approved:heavy_check_mark:
**Hi team!
Nice one! Thank you for implementing the changes requested by the last reviewer:clap::clap:. The game works as expected. The codes are logically divided between classes. Keep up the hard work:thumbsup:.
Move on with the next milestone:walking:.**

Kindly drop your comments/questions :point_down: and ping me on slack @ Eric Enaburekhan to respond.
Happy hacking:blush:
Eric Enaburekhan

**Pull request merged**